app/blog/2020-09-22-lit-element-and-lit-html-next-preview.md

```diff
@@ -0,0 +1,57 @@
+---
+title: "LitElement 3.0 & lit-html 2.0: Early Preview Release"
+---
+
+Today we're publishing the very firse preview releases of the next major versions of our flagship libraries, LitElement and lit-html.
+
+These releases are not yet feature complete or fully API stable, but they include most of the breaking changes that we intend to make, and most of the functionality we want to carry over from the previous versions. The releases also do not yet support legacy browsers like IE11, or browsers requiring the web components polyfills.
+
+## Motivation
+
+We've been very happy with the current version of our libraries - they're fast, small, and stable (yay!) - and in some ways we _don't_ have very many pressing needs to make breaking changes. We don't take making breaking changes lightly. But there are some compelling reasons for changes that we think will improve the user-experience of components and applications built with LitElement.
+
+1. Performance. We have found that some of our browser-bug workaround code and customization abstractions prevent optimizations that we would like to do.
+2. Size. That same code, and our extensive public API, costs bytes. We always want to find ways to make the liraries smaller.
+3. Features & API cleanup. Some features are difficult to add in a cost-effective way with the current architecture, or can't really be improved without small tecnically breaking change.
+4. Server-side-rendering: lit-html has an extremely flexible and customizable API, and in some ways is more of a template-system construction kit than a single template-system. But this flexibility complicates SSR, which needs to make assumptions about how the server-rendered HTML maps to templates. SSR would only work well with the default, uncustomized lit-html, so limiting customization makes SSR more reliable. Very few developers used the customization APIs anyway.
+
+   We also want to make the directive API SSR compatible by limiting access to the DOM.
+
+## What's changed
+
+These are new major versions, so there are some breaking changes, but we want to minimize disruption to our users as much as possible, so we've limited the breaking changes so that they don't affect most users, or only require a mechanical change (like changing an import).
```

Comment:
```suggestion
These are new major versions, so there are some breaking changes, but we want to minimize disruption to our users as much as possible. We've limited the breaking changes so that they don't affect most users, or only require a mechanical change (like changing an import).
```

Reply:
done